notebooks/MAST/TESS/beginner_tess_tap_search/beginner_tess_tap_search.ipynb

```diff
@@ -0,0 +1,375 @@
+    "print(footprintShape)\n",
+    "print(footprintVertices)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once the footprint has been isolated and reformatted, we perform another query listing all lightcurves (minus the data validation timeseries, whose files end in _dvt) by their target name, sector, RA and Dec, as well as returning the access url for each FITS file and its estimated file size. We are doing this as an asynchronous query, which handles longer response times, just in case."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "job = TAP_service.launch_job_async(\"\"\"\n",
+    "            SELECT target_name, sequenceNumber as sector, s_ra, s_dec, access_url, access_estsize, obs_id \n",
+    "            FROM dbo.caomobservation JOIN ivoa.obscore on dbo.caomobservation.observationID = ivoa.obscore.obs_id \n",
+    "            WHERE \n",
+    "            obs_collection = 'TESS' and dataproduct_type = 'timeseries' \n",
+    "            and access_url like '%lc.fits' and \n",
+    "            CONTAINS(POINT('ICRS',s_ra,s_dec),{}('ICRS', {}))=1\n",
+    "            ORDER BY obs_id\n",
+    "            \"\"\".format(footprintShape, footprintVertices))\n",
+    "\n",
+    "#uncomment the line below to see the whole query with wildcards replaced\n",
+    "#print(job.parameters)\n",
+    "TAP_results = job.get_results()\n",
+    "TAP_results          "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The returned data is in an astropy table; you can manipulate it to do more ordering or filtering. To download individual files or the whole set, you can use the access_url column, as below.\n",
+    "\n",
+    "Python's requests library lets you download files from a URL. The downloads will appear in the directory where your notebook is running."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Example: first result row:\n",
+    "single_url = TAP_results['access_url'][0].decode('UTF-8')\n",
+    "filename = TAP_results['obs_id'][0].decode('UTF-8') + \"_lc.fits\"\n",
+    "r = requests.get(single_url, allow_redirects=True)\n",
+    "open(filename, 'wb').write(r.content)\n",
+    "\n",
+    "\n",
+    "#Uncomment the code below to download every file in a loop. \n",
+    "#Warning: this can take some time as there are ~900 2megabyte files listed from the notebook's original sample query \n",
+    "#(see \"True length\" of masked table above).\n",
+    "\n",
+    "#for rows in TAP_results :\n",
+    "#    single_url = rows['access_url'].decode('UTF-8')\n",
+    "#    filename = rows['obs_id'].decode('UTF-8') + \"_lc.fits\"\n",
+    "#    r = requests.get(single_url, allow_redirects=True)\n",
+    "#    open(filename, 'wb').write(r.content)"
```

Comment:
Is there something that could be printed out in this cell to indicate that the download has completed?

Reply:
Good idea. Added file size from headers after download (and general 'all files complete' in the commented-out section.